src/main/java/ru/skypro/flea/service/impl/ImageServiceImpl.java

```diff
@@ -30,31 +32,27 @@ public ImageServiceImpl(String imagesPath) {
   public void checkCatalogue() {
     File catalogue = new File(imagesPath);
     if (!catalogue.exists()) {
-      boolean result = catalogue.mkdir();
+      boolean result = catalogue.mkdirs();
       if (result) {
-        log.info("Catalogue was created.");
+        log.info("Image resource catalogue was created.");
       } else {
-        log.info("Failed to create catalogue.");
-        throw new RuntimeException();
+        throw new FileSystemError("Failed to create image resource catalogue");
       }
     } else {
-      log.info("Catalogue already exists.");
+      log.info("Image resource catalogue already exists.");
     }
   }
 
   @Override
   public void saveImage(MultipartFile multipartFile, String fileName) {
     String extension = "";
-    String originalFileName = multipartFile.getOriginalFilename();
-    if (originalFileName == null) {
-      throw new RuntimeException();
-    }
-    int i = originalFileName.lastIndexOf(".");
+    String multipartFileName = multipartFile.getName();
```

Comment:
as it turned out we do not need originalFileName
and name can not be null
https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/multipart/MultipartFile.html#getName()

Reply:
nevermind...